### PR TITLE
[r] Rename SourceName to SourceSpec (#2843)

### PR DESCRIPTION
**Pull request by amarjandu**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in sandbox and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [x] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Review by danielsotirhos — CHANGES_REQUESTED** [review]
Found a couple more spots. 
[review.patch.txt](https://github.com/DataBiosphere/azul/files/6475433/review.patch.txt)

**Comment by amarjandu** [comment]
@hannes-ucsc from your original patch https://github.com/DataBiosphere/azul/issues/2650#issue-771289746 you requested that `.name` be changed to `.spec`, Was that modification to take place within the SourceRef? or within the subclasses of SourceName/SourceSpec?

If changing the attribute of `SourceName/SourceSpec` access would look like; `source.spec.spec` which is a little weird. Just want to clarify this before moving forward with the patch from @danielsotirhos

**Comment by hannes-ucsc** [comment]
When a variable or attribute references a value that is an instance of `SourceSpec`, that variable or attribute should be called `spec` or something derived from that, like `foo_spec` or `bar_spec`. When a variable or attribute references a value that is NOT an instance of `SourceSpec` (or something that wholly represents such an instance, like the JSON representation of such an instance), that variable or attribute should NOT be called `spec` or something derived from that, like `foo_spec` or `bar_spec`.

File: src/azul/indexer/__init__.py [hunk 1]
```diff
@@ -50,25 +50,23 @@ class BundleFQID(SupportsLessThan):
...
-#        https://github.com/DataBiosphere/azul/issues/2843
 @attr.s(frozen=True, auto_attribs=True, kw_only=True)
-class SourceName(ABC, Generic[SOURCE_NAME]):
+class SourceSpec(ABC, Generic[SOURCE_SPEC]):
     """
     The name of a repository source containing bundles to index. A repository
     has at least one source. Repository plugins whose repository source names
     are structured might want to implement this abstract class. Plugins that
-    have simple unstructured names may want to use :class:`StringSourceName`.
+    have simple unstructured names may want to use :class:`StringSourceSpec`.
```
Comment (danielsotirhos) [inline src/azul/indexer/__init__.py:62]:
```suggestion
    have simple unstructured names may want to use :class:`SimpleSourceSpec`.
```
There is no `StringSourceSpec`.

**Review by danielsotirhos — APPROVED** [review]
LGTM.
Has this been deployed to you personal deployment and reindexed to confirm things still work?

**Comment by amarjandu** [comment]
> LGTM.
> Has this been deployed to you personal deployment and reindexed to confirm things still work?

Been testing things out this morning, so far we look good... shall confirm before requesting primary review.

File: src/azul/indexer/document.py [hunk 2]
```diff
@@ -481,7 +481,7 @@ def to_json(self) -> JSON:
                     contents=self.contents)
 
     def to_dict(self) -> JSON:
-        return {f.name: getattr(self, f.name) for f in fields(self)}
+        return {f.spec: getattr(self, f.spec) for f in fields(self)}
```
Comment (hannes-ucsc) [inline src/azul/indexer/document.py:484]: Extremely worried why we didn't catch this either. Please find out why.

File: test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
             "entity_id": "aaa96233-bf27-44c7-82df-b4dc15ad4d9d",
             "source": {
                 "id": "4b737739-4dc9-5d4b-9989-a4942047c91c",
-                "name": "test"
+                "spec": "test"
```
Comment (hannes-ucsc) [inline test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json:11]: Can you post some evidence that in a real deployment's index, one that's populated from a TDR source, this is the full spec and not just the name?

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
It's possible that the assertion I added breaks things. In that case we should discuss possible remedies.

```diff
Index: src/azul/plugins/repository/dss/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/repository/dss/__init__.py b/src/azul/plugins/repository/dss/__init__.py
--- a/src/azul/plugins/repository/dss/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/plugins/repository/dss/__init__.py	(date 1621544428238)
@@ -78,13 +78,14 @@
     def for_dss_endpoint(cls, endpoint: str):
         # We hash the endpoint instead of using it verbatim to distinguish them
         # within a document, which is helpful for testing.
-        return cls(id=cls.id_from_name(endpoint),
-                   spec=SimpleSourceSpec(prefix=config.dss_query_prefix,
-                                         name=endpoint))
+        spec = SimpleSourceSpec(prefix=config.dss_query_prefix,
+                                name=endpoint)
+        return cls(id=cls.id_from_spec(spec),
+                   spec=spec)
 
     @classmethod
-    def id_from_name(cls, name: str) -> str:
-        return str(uuid5(cls.namespace, name))
+    def id_from_spec(cls, spec: SimpleSourceSpec) -> str:
+        return str(uuid5(cls.namespace, spec.name))
 
 
 DSSBundleFQID = SourcedBundleFQID[DSSSourceRef]
@@ -100,8 +101,8 @@
     def sources(self) -> AbstractSet[str]:
         return {config.dss_endpoint}
 
-    def lookup_source_id(self, name: SimpleSourceSpec) -> str:
-        return DSSSourceRef.id_from_name(name.name)
+    def lookup_source_id(self, spec: SimpleSourceSpec) -> str:
+        return DSSSourceRef.id_from_spec(spec)
 
     @cached_property
     def dss_client(self):
Index: src/azul/plugins/repository/tdr/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/repository/tdr/__init__.py b/src/azul/plugins/repository/tdr/__init__.py
--- a/src/azul/plugins/repository/tdr/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/plugins/repository/tdr/__init__.py	(date 1621544476934)
@@ -200,8 +200,8 @@
     def _assert_source(self, source: TDRSourceRef):
         assert source.spec in self.sources, (source, self.sources)
 
-    def lookup_source_id(self, name: TDRSourceSpec) -> str:
-        return self.tdr.lookup_source_id(name)
+    def lookup_source_id(self, spec: TDRSourceSpec) -> str:
+        return self.tdr.lookup_source_id(spec)
 
     def list_bundles(self, source: TDRSourceRef, prefix: str) -> List[TDRBundleFQID]:
         self._assert_source(source)
Index: src/azul/plugins/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/__init__.py b/src/azul/plugins/__init__.py
--- a/src/azul/plugins/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/plugins/__init__.py	(date 1621544795859)
@@ -193,9 +193,10 @@
 
     def resolve_source(self, *, spec: str, id: Optional[str] = None) -> SOURCE_REF:
         """
-        Return an instance of :class:`SourceRef` for the repository source with
-        the specified name or raise an exception if no such source exists. If an
-        ID is given, ensure that it refers to the same source as the name.
+        Return an instance of :class:`SourceRef` for the repository source
+        matching the given specification or raise an exception if no such source
+        exists. If an ID is given, ensure that the source matching the
+        specification has the given ID.
         """
         cls = type(self)
         base_cls = one(getattr(cls, '__orig_bases__'))
@@ -212,7 +213,7 @@
         return source_ref_cls(id=id, spec=spec)
 
     @abstractmethod
-    def lookup_source_id(self, name: SOURCE_SPEC) -> str:
+    def lookup_source_id(self, spec: SOURCE_SPEC) -> str:
         """
         Return the ID of the repository source with the specified name or raise
         an exception if no such source exists.
Index: src/azul/indexer/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/indexer/__init__.py b/src/azul/indexer/__init__.py
--- a/src/azul/indexer/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/indexer/__init__.py	(date 1621545731798)
@@ -86,6 +86,8 @@
         return cls(name=spec)
 
     def __str__(self) -> str:
+        # String representaton discards prefix so there mustn't be one
+        assert self.prefix is None, self.prefix
         return self.name
 
 
@@ -158,7 +160,7 @@
             else:
                 assert self.id == id
                 require(self.spec == spec,
-                        'Ambiguous source names for same ID.', self.spec, spec, id)
+                        'Ambiguous source specs for same ID.', self.spec, spec, id)
             return self
 
     def to_json(self):
Index: src/azul/indexer/document.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/indexer/document.py b/src/azul/indexer/document.py
--- a/src/azul/indexer/document.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/indexer/document.py	(date 1621545240240)
@@ -481,7 +481,7 @@
                     contents=self.contents)
 
     def to_dict(self) -> JSON:
-        return {f.spec: getattr(self, f.spec) for f in fields(self)}
+        return {f.name: getattr(self, f.name) for f in fields(self)}
 
     @classmethod
     def _from_json(cls, document: JSON) -> Mapping[str, Any]:
Index: src/azul/plugins/metadata/hca/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/metadata/hca/__init__.py b/src/azul/plugins/metadata/hca/__init__.py
--- a/src/azul/plugins/metadata/hca/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/plugins/metadata/hca/__init__.py	(date 1621544096433)
@@ -186,7 +186,7 @@
                 "entryId": "entity_id",
 
                 "sourceId": "sources.id",
-                "sourceSpec": "sources.name",
+                "sourceSpec": "sources.spec",
             },
             autocomplete_translation={
                 "files": {
```

File: src/azul/plugins/metadata/hca/__init__.py [hunk 4]
```diff
@@ -186,7 +186,7 @@ def service_config(self) -> ServiceConfig:
                 "entryId": "entity_id",
 
                 "sourceId": "sources.id",
-                "sourceName": "sources.name",
+                "sourceSpec": "sources.name",
```
[the remark this answers is not in the document]
Comment (amarjandu) [reply src/azul/plugins/metadata/hca/__init__.py:189]: https://github.com/DataBiosphere/azul/pull/3047#issuecomment-844351231 
ran both before requesting review, ill dig into this, i thought i caught all of these renames.

File: test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json (hunk 3 above)
Comment (amarjandu) [reply test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json:11]: I selected the dev deployment and used the kibana proxy to show the value in the dcp2 bundles index.
Hopefully this is sufficient, bottom of the image:

Screen Shot 2021-05-20 at 4 25 22 PM

File: src/azul/indexer/document.py (hunk 2 above)
Comment (amarjandu) [reply src/azul/indexer/document.py:484]: delete as first commit and run tests

File: src/azul/plugins/metadata/hca/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (amarjandu) [reply src/azul/plugins/metadata/hca/__init__.py:189]: add fixme "Build does not detect invalid values in ServiceConfig.translation" ln 125

[the remark this answers is not in the document]
Comment (amarjandu) [reply src/azul/plugins/metadata/hca/__init__.py:189]: test w/ sort by sourceSpec

File: src/azul/indexer/document.py (hunk 2 above)
Comment (amarjandu) [reply src/azul/indexer/document.py]: Unit tests pass + IT passes with this method removed

File: src/azul/plugins/metadata/hca/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (amarjandu) [reply src/azul/plugins/metadata/hca/__init__.py:191]: raised #3071 to track this.

**Comment by amarjandu** [comment]
```
diff --git a/src/azul/indexer/__init__.py b/src/azul/indexer/__init__.py
--- a/src/azul/indexer/__init__.py	(revision 13c883521ac9f81ebc9a328a782e93f9ef915dfb)
+++ b/src/azul/indexer/__init__.py	(date 1621545731798)
@@ -86,6 +86,8 @@
         return cls(name=spec)
 
     def __str__(self) -> str:
+        # String representaton discards prefix so there mustn't be one
+        assert self.prefix is None, self.prefix
         return self.name
```

This assertion is breaks IT.

File: src/azul/indexer/__init__.py [hunk 5]
```diff
@@ -83,10 +83,43 @@ class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
 
     @classmethod
     def parse(cls, spec: str) -> 'SimpleSourceSpec':
-        return cls(name=spec)
+        """
+        >>> s = SimpleSourceSpec.parse('https://foo.edu:12')
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:87]: Assignment to `s` is unnecessary.

File: src/azul/indexer/__init__.py [hunk 6]
```diff
@@ -83,10 +83,43 @@ class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
...
+        >>> s
+        SimpleSourceSpec(prefix='12', name='foo:bar/baz')
+
+        >>> s = SimpleSourceSpec.parse('foo')
+        >>> s
+        SimpleSourceSpec(prefix='', name='foo')
+        """
+        # FIXME: Move parsing of prefix to SourceSpec
+        #        https://github.com/DataBiosphere/azul/issues/3073
+        if ':' in spec:
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:101]: The double lookup can be eliminated:

image

Comment (hannes-ucsc) [reply src/azul/indexer/__init__.py:101]: Anyways, the colon shouldn't be optional. But the condition is violating the symmetry invariant I mentioned in our convo.

File: src/azul/indexer/__init__.py [hunk 7]
```diff
@@ -83,10 +83,43 @@ class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
...
+        'foo:bar/baz:12'
+
+        >>> s = SimpleSourceSpec(prefix=':12', name='foo:bar/baz')
+        >>> str(s)
+        Traceback (most recent call last):
+        ...
+        AssertionError
+
+        """
+        assert ':' not in self.prefix
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:121]: Same assert wouldn't hurt in `parse`

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: src/azul/indexer/__init__.py (hunk 7 above)
Comment (amarjandu) [reply src/azul/indexer/__init__.py:121]: I think doing some type of UUID validation would be good to prevent stuff like;
```
SimpleSourceSpec.parse('https://amar.edu')
SimpleSourceSpec(prefix='//amar.edu', name='https')
```

Comment (amarjandu) [reply src/azul/indexer/__init__.py:121]: As an alternative to both assertions; why not add a attrs validator to the prefix attribute?

```
@attr.s(frozen=True, auto_attribs=True, kw_only=True)
class SourceSpec(ABC, Generic[SOURCE_SPEC]):

    prefix: str = attr.ib(default='',
                          validator=validate_uuid)
```

We are ensured our prefixes are valid when the objects are instantiated, and we can remove the assertions in both locations, as it would fail due to the prefix being invalid w/ a `:` present.

Comment (hannes-ucsc) [reply src/azul/indexer/__init__.py:121]: Assertions and validations are different things. The former express program invariants, the latter express constraints on program input. We'd still want the assertion, even if we did a validation. I would write a `__post_init` (less code that a validator) and in it, require that the prefix is a valid UUID prefix and then assert that it doesn't have a colon in it.

File: src/azul/indexer/__init__.py [hunk 8]
```diff
@@ -64,6 +68,10 @@ class SourceSpec(ABC, Generic[SOURCE_SPEC]):
 
     prefix: str = ''
 
+    def __attrs_post_init__(self):
+        assert ':' not in self.prefix
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:72]: I want you to think about the ordering of these two lines and tell me what might be wrong here.

File: src/azul/indexer/__init__.py [hunk 9]
```diff
@@ -84,41 +92,37 @@ class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
...
         """
         # FIXME: Move parsing of prefix to SourceSpec
         #        https://github.com/DataBiosphere/azul/issues/3073
-        if ':' in spec:
-            name, _, prefix = spec.rpartition(':')
-        else:
-            name = spec
-            prefix = ''
+        name, sep, prefix = spec.rpartition(':')
+        if not sep:
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:112]: https://github.com/DataBiosphere/azul/pull/3047#discussion_r640200730

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: src/azul/indexer/__init__.py (hunk 8 above)
Comment (amarjandu) [reply src/azul/indexer/__init__.py]: If we change the order:
the presence of a `:` in the prefix will cause `validate_uuid_prefix` to raise an InvalidUUIDPrefix error, and we would never reach the following `assert`. 

If we don't change the order:
The assertion can be used to express the invariant that a prefix does not contain a `:`, if that is asserted, we can inspect if the uuid prefix is valid.

There might be something I'm missing here.

File: src/azul/indexer/__init__.py [hunk 10]
```diff
@@ -77,25 +83,62 @@ def __str__(self) -> str:
...
+        >>> SimpleSourceSpec.parse('https://foo.edu')
+        Traceback (most recent call last):
+        ...
+        azul.uuids.InvalidUUIDPrefixError: '//foo.edu' is not a valid UUID prefix.
+        """
+
+        # FIXME: Move parsing of prefix to SourceSpec
+        #        https://github.com/DataBiosphere/azul/issues/3073
+        name, sep, prefix = spec.rpartition(':')
+        require(bool(sep), 'Source specs must contain a prefix delimiter')
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:120]:
```suggestion
        reject(sep is None, 'Source specifications must end in a colon followed by an optional UUID prefix}')
```

potentially wrapped.

File: src/azul/indexer/__init__.py [hunk 11]
```diff
@@ -77,25 +83,62 @@ def __str__(self) -> str:
...
+        return cls(prefix=prefix, name=name)
 
     def __str__(self) -> str:
-        return self.name
+        """
+        >>> str(SimpleSourceSpec(prefix='12', name='foo:bar/baz'))
+        'foo:bar/baz:12'
+
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:127]: This method is trivial and doesn't need a doctest.

File: src/azul/indexer/__init__.py [hunk 12]
```diff
@@ -77,25 +83,62 @@ def __str__(self) -> str:
...
+        >>> str(SimpleSourceSpec(prefix=':12', name='foo:bar/baz'))
+        Traceback (most recent call last):
+        ...
+        azul.uuids.InvalidUUIDPrefixError: ':12' is not a valid UUID prefix.
```
Comment (hannes-ucsc) [inline src/azul/indexer/__init__.py:131]: The exception is not raised by this method so the test is not relevant here.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Nits.

File: src/azul/indexer/__init__.py (hunk 10 above)
Comment (amarjandu) [reply src/azul/indexer/__init__.py]: https://docs.python.org/3/library/stdtypes.html#str.rpartition
```
 If the separator is not found, return a 3-tuple containing two empty strings, followed by the string itself.
```

Going to use condition `sep is ''`

**Review by hannes-ucsc — APPROVED** [review]

**Comment by amarjandu** [comment]
@hannes-ucsc 
There was a failure in the unittests for the sandbox see https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/21920#L5967.

The sandbox still has `AZUL_DSS_QUERY_PREFIX` being set https://github.com/DataBiosphere/azul/blob/develop/deployments/sandbox/environment.py#L99 which causes the ES index to use it when creating the SourceSpec for the `test` catalog.
Screen Shot 2021-06-10 at 4 26 41 PM

We can also patch/modify the results of the canned bundled to match the SourceSpec of each deployment.

Might be a good time to bring up https://github.com/DataBiosphere/azul/issues/2667, with a parent deployment being shared with the sandbox, this might have been caught earlier.

**Comment by amarjandu** [comment]
Pulling this, need to update to include CannedStagingArea plugin as well...

**Comment by amarjandu** [comment]
@hannes-ucsc please note the additional changes to the canned repository plugin.

File: deployments/sandbox/environment.py [hunk 13]
```diff
@@ -96,8 +96,6 @@ def env() -> Mapping[str, Optional[str]]:
             }
         ),
 
-        'AZUL_DSS_QUERY_PREFIX': '42',
```
Comment (hannes-ucsc) [inline deployments/sandbox/environment.py:99]: What does this have to do with #2843, the issue referenced in the title of the commit that this change will be squashed into? AFAICT, this is a left-over from an already merged PR, so this change should be in a commit whose title references the issue that other PR addressed.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: test/service/test_response.py [hunk 14]
```diff
@@ -821,7 +821,7 @@ def test_projects_key_search_response(self):
                     ],
                     "sources": [{
                         "sourceId": "4b737739-4dc9-5d4b-9989-a4942047c91c",
-                        "sourceName": "test"
+                        "sourceSpec": "test:"
```
Comment (hannes-ucsc) [inline test/service/test_response.py:824]: @noah-aviel-dove FYI

**Review by hannes-ucsc — APPROVED** [review]

**Comment by jessebrennan** [comment]
Failures after reindexing dev:
```
40733888-3b9f-500e-84fd-1a18a64aadbe
94565940-7b2f-5fd7-beb1-dc309687fe09
9d0f5cd1-0ef6-5eed-8dd7-f15d36a34185
ae338c4e-650f-545b-8631-3c8d755aa2f9
c12a6ca2-3234-5182-ad0b-f8c2fb881ab5
```
Which match the expected failures from #2870 and #2873.
